### ISSUE TITLE
Compiler warnings in main_into_hybrid

### ISSUE THREAD

**Issue by spradlin** [spradlin/WCSim#16]
Compilation warnings may indicate unwanted behavior, point to sources of inefficiency, and mask more serious problems. Efforts should be made to eliminate them.

This issue refers to the branch [tdealtry/main_into_hybrid](https://github.com/tdealtry/WCSim/tree/main_into_hybrid), whiich is the merge candidate that will bcome the head of the repository after finalization.  My previous Issue #9 should be considered obsolete, but i do want to revisit some of the non-trivial issues identified in that issue.

I am using the `cmake`-based build in a docker container based on the one produced by [main_into_hybrid/Dockerfile.base](https://github.com/tdealtry/WCSim/blob/main_into_hybrid/Dockerfile.base).

The stdout+stderr of the build process for the head of [tdealtry/main_into_hybrid](https://github.com/tdealtry/WCSim/tree/main_into_hybrid) as of Wed Jan  4 15:02:20 UTC 2023 is in the attached file.
[make.log](https://github.com/spradlin/WCSim/files/10345159/make.log)

**Comment by spradlin** [comment]
## Arguments shadowing data members

As as noted in the previous Issue #9. a number of the warnings have the form
> `warning: declaration of ‘<variable name>’ shadows a member of 'this'`

I will adopt the same strategy that is described [in that issue](https://github.com/spradlin/WCSim/issues/9#issuecomment-1001839300) and reproduced below.

These often occur in class constructors, where the value of the argument that shadows the data member is used to initialize the data member.  In these cases, the end result is often what is desired, despite the warning.  However, it is easy to create instances that have undesired effects, even in constructors, especially if the argument is passed by reference.  This kind of pattern should really be avoided.

Many large toolkits avoid such problems by adhering to naming conventions for data members and arguments.  For example, GEANT4 appears to conform to a convention in which every class private data member has a name that begins with `f`.  *For classes in which i encounter a shadowing warning for a private data member, i will change the names of the private data members to begin with `f`, as GEANT4 does.*

In some cases, an argument name shadows a **public** data member.  For these cases, i will chang the arguments' names, either by abbreviating them or expanding them.

**Comment by spradlin** [comment]
## Warnings from WCSimConstructCylinder.cc

The first group of compilation warnings is from [`src/WCSimConstructCylinder.cc`](https://github.com/tdealtry/WCSim/blob/main_into_hybrid/src/WCSimConstructCylinder.cc).  As there [were in the WCSim/develop branch](https://github.com/spradlin/WCSim/issues/9#issuecomment-1002371004), there are a large number of warnings in this group, and they generally fall into two categories:
1. Local variables that shadow  variables in a containing scope (see [message above](https://github.com/spradlin/WCSim/issues/16#issuecomment-1371055373)),
2. Unused variables that just store copies of pointers to newly created geometry objects that are owned, presumably, by GEANT4's geometry manager.

The second class are usually easier to address, so let us start there.

### Unused variables in WCSimConstructCylinder.cc

With links to the original lines of source code.  Unless otherwise noted, the declaration of the unused variable is commented out.  I do not like to leave commented code, but the variable names are generally meaningful labels that are usefully kept, for now.
- [Line 1361](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1361):  `physiTopCapWLSPlate`
- [Line 1372](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1372):  `physiBottomCapWLSPlate`
- [Line 1090](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1090):  `physiWCODTopCapsTyvek`
- [Line 1100](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1100):  `WaterTySurfaceTop`
- [Line 1105](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1105):  `physiWCODBottomCapsTyvek`
- [Line 1151](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1151):  `physiWCBarrelCellODTyvek`
- [Line 1160](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1160):  `WaterTySurfaceSide`
- [Line 236](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L236):  `physiWCBarrel`
- [Line 367](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L367):  `physiWCBarrelAnnulus`
- [Line 400](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L400):  `physiWCBarrelRing`
- [Line 507](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L507):  `WaterBSBarrelCellSurface`
- [Line 548](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L548):  `logicWCExtraBorderCell`.  This line was removed.  There is a later declaration of a variable with the same name at [line 1586](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1586) that *is* used.
- [Line 1408](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1408):  `physiTopCapAssembly`
- [Line 1417](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1417):  `physiBottomCapAssembly`
- [Line 1643](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1643):  `WaterBSExtraBorderCellSurface`
- [Line 1955](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1955):  `physiCapPMT`.  Note that this variable name does appear in comments.

**Comment by spradlin** [comment]
### Shadowing warnings in WCSimDetectorConstruction::ConstructCylinder (WCSimConstructCylinder.cc)

- `WCPMTRotation` at [line 1004](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1004) and at [line 1289](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1289) shadow declaration at [line 936](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L936) in a containing scope.  The reuse of this variable name is more a product of poor encapsulation than anything else.  Each occurrance includes a creation of a `new` `G4RotationMatrix` on the heap that is used exactly once in a `G4PVPlacement`.
  - Do the objects pointed to by `WCPMTRotation` need to be created on the heap?  Maybe.  The [`G4PVPlacement` constructor](https://gitlab.cern.ch/geant4/geant4/-/blob/geant4-10.3-release/source/geometry/volumes/include/G4PVPlacement.hh#L55-62) explicitly requires a `G4RotationMatrix *` or a [`const G4Transform3D &`](https://gitlab.cern.ch/geant4/geant4/-/blob/geant4-10.3-release/source/geometry/volumes/include/G4PVPlacement.hh#L77-83) that is transformed to a `G4RotationMatrix *` [newly created on the heap](https://gitlab.cern.ch/geant4/geant4/-/blob/geant4-10.3-release/source/geometry/volumes/src/G4PVPlacement.cc#L392-407).  Reuse of the instance is probably precluded.
  - What is the potential for confusion if the pointer variable is reused without redeclaration?  Pretty high.  These are three different rotation matrixes.
  -  **To avoid confusion, these variables should have different names.**  
  - The original instance at [line 936](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L936) is for placing a PMT within the `logicWCBarrelCell`.  Leave this name unchanged.
  - The second at [line 1004](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1004) is for placing an ID PMT within the `logicWCExtraTowerCell`.  Rename it to `WCExtraPMTRotation` and update its usage in lines [1005-1006](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1005-L1006) and [1024](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1024).
  - The third at [line 1289](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1289) is for placing an OD PMT within the `logicWCExtraTowerCell`.  Rename it to `WCExtraODPMTRotation` and update its usage in lines [1290-L1291](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1290-L1291) and [1310](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1310).
- `horizontalSpacing` at [line 1010](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1010) shadows declaration at [line 955](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L955) in a containing scope.  The instance at [line 1010](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1010) is used in the construction of the 'extra' tower cell.  The two instances are different values and easily confused.  Rename the instance specific to the 'extra' cell `horizontalSpacingExtra` and update its use at line [1016](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1016).
- `verticalSpacing` at [line 1011](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1011) shadows declaration at [line 956](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L956) in a containing scope.  The issue is the same as previous for `horizontalSpacing`:  the shadowing instance is for the 'extra' cell.  However, in this case the two instances appear to have identical values, which makes sense for this case and appears to be the intent.  In this case, it makes sense to have the code reflect that the two values are supposed to be identical by eliminating the definition at [line 1011](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1011) so that the one at [line 956](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L956) is the only one.  I will leave a comment in the code to make this clearer.
- `sphereRadius` at [line 1059](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1059) shadows declaration at [line 114](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L114).  This is interesting.  The two instances of `sphereRadius` have identical values, and this seems to be the intent.  `sphereRadius` is only ever used as an intermediate term to calculate barrel radii in the presence of OD:  `outerAnnulusRadius` in [L118-L121](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L118-L121) and `WCODRadius` in [L1061](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1061).  Both of these are data members of `WCSimDetectorConstruction` and have a scope external to this function that calculates its values.
  - Duplication of the calculation of `sphereRadius` is error-prone
  - Especially with the instances to far apart.
  - Its value should be calculated *only once*.  To facilitate this, i propose to eliminate the copy at [line 1059](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1059) and move the calculation of `WCODRadius` from [L1061](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1061) to the just after the `outerAnnulusRadius` computation in [L118-L121](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L118-L121).  I also want to rearrange the calculations a bit to make the relationships of the dimensions a little clearer.
- `horizontalODSpacing` at [L1297](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1297) shadows declaration at [L1208](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1208) in a containing scope.  The relationship between the two instances is the same as that for `horizontalSpacing` as described above: they are two different values, one used in constructing a standard cell and one for constructing an 'extra' cell.  Rename the instance at [L1297](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1297) to `horizontalODSpacingExtra` and update its use on [L1304](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1304).
- `verticalODSpacing` at [L1298](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1298) shadows declaration at [L1209](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1209) in a containing scope.  Like the two instances of `verticalSpacing` above, the two instances of `verticalODSpacing` have deliberately identical values.  Remove the instance at [L1298](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1298) and leave a comment to make the equality clear.

**Comment by spradlin** [comment]
### Shadowing warnings in WCSimDetectorConstruction::ConstructCaps (WCSimConstructCylinder.cc)

- `WCPMTRotation` at lines [L2046](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2046) and [L2164](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2164) shadow declaration at [L1994](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1994).  This set of overloaded variable names are analagous to the previously set of the same name.  This set is in another function and does not interfere with the previous set.  Adopt the same set of solutions:
  - The original instance at [L1994](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1994) is for placing PMTs in `logicWCBarrelBorderCell`.  Leave its name unchanged.
  - The instance at [L2046](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2046) is for placing ID PMTs in `logicWCExtraBorderCell`.  Rename it to `WCExtraPMTRotation` and update its usage at [L2047-L2048](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2047-L2048) and [L2065](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2065).
  - The instance at [L2164](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2164) is for placing OD PMTs in `logicWCExtraBorderCell`.  Rename it to `WCExtraODPMTRotation` and update its use in [L2165-L2166](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2165-L2166) and [L2185](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2185).
- `horizontalSpacing` at [L2052](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2052) shadows instance at [L2004](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2004) in containing scope.  The situation is the same as that for the variable of the same name in `WCSimDetectorConstruction::ConstructCylinder()`, and will solve in the same way.   Rename the instance specific to the 'extra' cell `horizontalSpacingExtra` and update its use in [L2058](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2058).
- `verticalSpacing` in [L2053](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2053) shadows declaration [L2005](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2005).  The situation is the same as that for the variable of the same name in `WCSimDetectorConstruction::ConstructCylinder()`---these are the same value by design and there should be only one representation of it.  Eliminate the instance in [L2053](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2053) and leave an explanatory comment.
- `barrelODCellWidth` in [L2161)](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2161) shadows declaration at [L2122](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2122).  The values of the two instances are identical, and appear to be so by design.  This is another parameter that is reused in construction of an 'extra' cell.  Eliminate the instance in [L2161)](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2161) and leave a comment.
- `barrelODCellHeight` at [L2162](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2162) shadows declaration at [L2123](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2123).  The values of the two instances are identical, and appear to be so by design.  This is another parameter that is reused in construction of an 'extra' cell.  Eliminate the instance in [L2162](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2162) and leave a comment.
- `horizontalODSpacing` in [L2172](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2172) shadows instance in [L2124](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2124).  The situation is the same as that for the variable of the same name in `WCSimDetectorConstruction::ConstructCylinder()`, and will solve in the same way.   Rename the instance specific to the 'extra' cell on [L2172](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2172) `horizontalODSpacingExtra` and update its use in [L2179](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2179).
- `verticalODSpacing` in [L2173](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2173) shadows declaration in [L2125](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2125).  The situation is the same as that for the variable of the same name in `WCSimDetectorConstruction::ConstructCylinder()`---these are the same value by design and there should be only one representation of it.  Eliminate the instance in [L2173](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2173) and replace with an explanatory comment.

**Comment by spradlin** [comment]
### Unitialized variables in WCSimConstructCylinder.cc

A few lines of code cause '<variable name> may be used uninitialized' warnings.  If there are conditions under which the unitialized variables are used, this is a problem.
- `logicWCPMT2` used in function call [L1956-L1963](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1956-L1963).  The variable is declared and initialized in [L1885-L1889](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1885-L1889).  The issue seems to be that the variable is initialized only if the variable `hybrid` evaluates to `true` in [L1886](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1886).  This suggests that it should only be used under this condition.  The conditional in [L1958](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1958) does have `hybrid` as a predicate to using `logicWCPMT2`, but, wow, this looks fragile.  For now, initialize it to `nullptr` at declaration, but **the logic here should be revisited**.
- `logicWCExtraBorderCell` used in function call [L2185-L2192](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2185-L2192).  This is interesting.  The variable is declared at [L1586](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1586).  It is initialized at [L1610-L1614](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1610-L1614) within a conditional block [L1588-L1649](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1588-L1649).  It is subsequently used at [L1620](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1620), [L1628-L1629](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1628-L1629), and [L1638](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1638) within that conditional block.  It is also used at [L2069](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2069), but this usage does not trigger an 'uninitialized' warning.  Why?  It is in a `#ifdef` preprocessor conditional [L2062-L2073](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2062-L2073), but [L64-L66](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L64-L66) indicate that this condition should be satisfied.  Perhaps because it is in the conditional block [L2040-L2081](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2040-L2081) that has the same conditional as the block that contains the initialization, [L1588](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1588)?  But this looks like the same conditional as [L2149-L2197](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2149-L2197) that contain the line that raises the warning.  Even more confusing, the usage at [L2152-L2159](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L2152-L2159) is also in this conditional block without an accompanying warning.  Why the warning is raised specifically here, but i do not see any usage outside of a conditional equivalent to that which contains initialization.  I will initialize it to `nullptr` at declaration, but **the logic must be revisited**.
- `logicWCPMT2` used in function call [L910-L917](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L910-L917).  The variable is declared at [L848](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L848) and initialized in the conditional block [L852-L860](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L852-L860).  This issue looks to be the same as that for the variable of the same name in `WCSimDetectorConstruction::ConstructCaps()`, which is described above.  Address it in the same way:  initialize to `nullptr` at declaration and **revisit the logic of this mess** later.
- `logicWCTopVeto` used in function call [L910-L917](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L910-L917).  It is declared at [L716](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L716).  It is initialized at [L732-L736](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L732-L736) within the conditional block [L718-L821](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L718-L821).  The conditional initialization depends on `WCTopVeto` evaluating to `true`.  This is the same as the condition of the block [L882-L928](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L882-L928) that contains the usage that raises the warning.  It do not see a case in which this is actually being used uninitialized.  It should be safe to initialize it to `nullptr` at declaration, but **the logic must be revisited**.
- `logicWCExtraTowerCell` used in function call [L1310-L1317](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1310-L1317).  It is declared in `WCSimDetectorConstruction::ConstructCylinder()` at [L547](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L547).  It is initialized at [L609-L613](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L609-L613) within the conditional block [L549-L708](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L549-L708).  The usage that throws the warning is within a block with the same condition [L1243-L1323](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructCylinder.cc#L1243-L1323).  It do not see a case in which this is actually being used uninitialized.  It should be safe to initialize it to `nullptr` at declaration, but **the logic must be revisited**.

**Comment by spradlin** [comment]
## Warnings from WCSimConstructGeometryTables.cc

I see one compiler warning from [`src/WCSimConstructGeometryTables.cc`](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructGeometryTables.cc) regarding the potential unitialized usage of `cylLocation` at [L368-L377](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructGeometryTables.cc#L368-L377).  `cylLocation` is declared at [L260](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructGeometryTables.cc#L260).

It is never formally initialized.  It serves as a kind of auxilliary loop variable that is periodically updated within the loop according to specific conditions.  It is recycled and used in three successive loops:  [L273-L320](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructGeometryTables.cc#L273-L320), [L331-L380](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructGeometryTables.cc#L331-L380), and [L391-L437](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructGeometryTables.cc#L391-L437).

It is unitialized at the start of the first loop, [L273-L320](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructGeometryTables.cc#L273-L320).  It is only initialized on the first iteration of the loop that satisfies the condition in [L283](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructGeometryTables.cc#L283).  Its value is updated on subseqent iterations of the loop that satisfy the condition.  It is used in the function call [L307-L316](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructGeometryTables.cc#L307-L316) on *every* iteration, regardless of whether it has been initialized on this or a previous iteration.

At the beginning of the second loop [(L331-L380)](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructGeometryTables.cc#L331-L380), `cylLocation` has whatever value it had at the end of the first loop, or it is unitialized if the condition on [L283](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructGeometryTables.cc#L283) was never satisfied.  It is updated/reinitialized on the first iteration of the second loop that satisfies [L341)](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructGeometryTables.cc#L341), and updated on every subsequent iteration that satisfies this condition.  It is used in the function call [L368-L377](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimConstructGeometryTables.cc#L368-L377) on *every* iteration, regardless of whether it has been initialized in the first loop or on this or a previous iteration of the second loop.

The situation of the third loop is the same as that for the second.

It may be that the condition within each loop for initializing/updating is always satisfied on the first iteration.  However, such a feature is not intrinsic to the function `WCSimDetectorConstruction::DumpGeometryTableToFile()`.

It would be better to initialize `cylLocation` to an easily recognized invalid value before the beginning of each loop.  This serves the dual purposes of addressing the 'uninitialized' warning and providing output that clearly indicates when the assumptions encoded in the conditions are bad.  I added a `cylLocation = -1` statement immediately before the start of each loop.

**Comment by spradlin** [comment]
## Warnings from WCSimEventAction.cc

```
/opt/HyperK/WCSim/src/WCSimEventAction.cc: In member function 'virtual void WCSimEventAction::EndOfEventAction(const G4Event*)':
/opt/HyperK/WCSim/src/WCSimEventAction.cc:262:42: warning: 'run' may be used uninitialized in this function [-Wmaybe-uninitialized]
       GetRunAction()->EndOfRunAction(run);
                                          ^
```

In `WCSimEventAction::EndOfEventAction()`, the complaint arises from the [L262](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimEventAction.cc#L262) use of the variable `run`, which is declared without initialization in the previous line, [L261](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimEventAction.cc#L261).  This is part of a hard program termination condition [L260-L264](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimEventAction.cc#L260-L264).  It has a normal exit status (0, [L263](https://github.com/tdealtry/WCSim/blob/e2a7d618f4723fd4db68a5be4016adfd3f030e75/src/WCSimEventAction.cc#L263) ).  This is weird for what is essentially an abort.

This abort is not part of the [current WCSim/develop branch](https://github.com/WCSim/WCSim/blob/ff3c12b88117741ab305d1f693590522ac19477a/src/WCSimEventAction.cc).  Where did this come from, and why was it put in?
- It is present in the [nuPRISM/develop branch](https://github.com/nuPRISM/WCSim/blob/6deb9050bb57d655bd530eab4d4b29a388f806b5/src/WCSimEventAction.cc)  [WCSimEventAction.cc#L215-L219](https://github.com/nuPRISM/WCSim/blob/6deb9050bb57d655bd530eab4d4b29a388f806b5/src/WCSimEventAction.cc#L215-L219).
- In the nuPRISM/develop branch, it was introduced in [commit 7b37f5c](https://github.com/nuPRISM/WCSim/commit/7b37f5cdc150d4655b65a60860bee17428237b48), ['Merge nuPRISM/mPMT with nuPRISM/devel now and solved all conflict.'](https://github.com/nuPRISM/WCSim/commit/7b37f5cdc150d4655b65a60860bee17428237b48) on Feb 24, 2017.
- The trail is cold at this point.  The merged branch nuPRISM/mPMT mentioned in the commit message is no longer present in the repository.

This doesn't make sense to me.  Let me ask Tom.

**Comment by spradlin** [comment]
## Warnings from WCSimPMTObject.cc

All of the warnings from [`src/WCSimPMTObject.cc`](https://github.com/tdealtry/WCSim/blob/0b06acf4572ade3915e697c8fec5b87d8ebdb723/src/WCSimPMTObject.cc) relate to unused parameters of the `HitTimeSmearing()` class member functions of selected PMT-type-specific specializations of the `WCSimPMTObject` class.  The base class member function [`WCSimPMTObject::HitTimeSmearing(double, double)`](https://github.com/tdealtry/WCSim/blob/0b06acf4572ade3915e697c8fec5b87d8ebdb723/include/WCSimPMTObject.hh#L24) is declared as pure virtual, so every class that derives from it must define a function with the same name and signature.

It looks like the [WCSim/develop version of the same virtual base class function](https://github.com/WCSim/WCSim/blob/ff3c12b88117741ab305d1f693590522ac19477a/include/WCSimPMTObject.hh#L23) took a single argument.  It looks like many of these 'unused parameter' warnings result from adapting PMT types that are only defined in  [WCSim/develop](https://github.com/WCSim/WCSim/blob/ff3c12b88117741ab305d1f693590522ac19477a/include/WCSimPMTObject.hh) being updated to the two-parameter interface.

To see how the two-parameter implementation relates to the single-parameter implementation, let me look at a PMT type that is defined otherwise identically in the [WCSim/develop](https://github.com/WCSim/WCSim/blob/ff3c12b88117741ab305d1f693590522ac19477a/src/WCSimPMTObject.cc) and [nuPRISM/develop](https://github.com/nuPRISM/WCSim/blob/6deb9050bb57d655bd530eab4d4b29a388f806b5/src/WCSimPMTObject.cc) versions.  I would expect the [`PMT20inch` default type of the `SuperK` geometry](https://github.com/tdealtry/WCSim/blob/0b06acf4572ade3915e697c8fec5b87d8ebdb723/src/WCSimDetectorConfigs.cc#L25) to be consistent.  Conveniently, this is also a type that does not produce an 'unused parameter' warning.

The WCSim/develop version definition of `PMT20inch` is [L75-L249 of src/WCSimPMTObject.cc](https://github.com/WCSim/WCSim/blob/ff3c12b88117741ab305d1f693590522ac19477a/src/WCSimPMTObject.cc#L75-L249).  Specifically, the one-parameter definition of [`PMT20inch::HitTimeSmearing(double Q)`](https://github.com/WCSim/WCSim/blob/ff3c12b88117741ab305d1f693590522ac19477a/src/WCSimPMTObject.cc#L82-L89) is
```
double PMT20inch::HitTimeSmearing(double Q) {
  double timingConstant = 10.0;
  double timingResolution = 0.33 + sqrt(timingConstant/Q);
  // looking at SK's jitter function for 20" tubes
  if (timingResolution < 0.58) timingResolution=0.58;
  double Smearing_factor = G4RandGauss::shoot(0.0,timingResolution);
  return Smearing_factor;
}
```

The nuPRISM/develop version definition of `PMT20inch` is [L78-L269 of src/WCSimPMTObject.cc](https://github.com/nuPRISM/WCSim/blob/6deb9050bb57d655bd530eab4d4b29a388f806b5/src/WCSimPMTObject.cc#L78-L269).  Specifically, the two-parameter definition of [`PMT20inch::HitTimeSmearing(double Q, double TTSFF=1.0)`](https://github.com/nuPRISM/WCSim/blob/6deb9050bb57d655bd530eab4d4b29a388f806b5/src/WCSimPMTObject.cc#L85-L106) is
```
double PMT20inch::HitTimeSmearing(double Q, double TTSFF=1.0) {
  double timingConstant = 10.0;
  double timingResolution;
  double Smearing_factor;
//if (linearity == 0){
    timingResolution = 0.33 + sqrt(timingConstant/Q);
    timingResolution *= TTSFF;
    // looking at SK's jitter function for 20" tubes
    if (timingResolution < 0.58) timingResolution=0.58;
    Smearing_factor = G4RandGauss::shoot(0.0,timingResolution);
    return Smearing_factor;
//}
//else {
//  timingResolution = 3.5
//  timingResolution *= TTSFF;
//  // looking at SK's jitter function for 20" tubes
//  if (timingResolution < 0.58) timingResolution=0.58;
//  Smearing_factor = G4RandGauss::shoot(0.0,timingResolution);
//  return Smearing_factor;
//}
    
}
```

Ignoring the vestigial commented code in the nuPRISM/develop version, the difference is that the `timingResolution` is multiplied by the new second argument before capping `timingResolution`.  Sadly, i suspect that the `SFF` in `TTSFF` stands for 'Smearing Fudge Factor'.  

The only instances of calls to `HitTimeSmearing` functions that i can find in the `tdealtry/main_into_hybrid` code base are in the `WCSimWCDigitizerSKI::DigitizeHits()` function at [L355](https://github.com/tdealtry/WCSim/blob/0b06acf4572ade3915e697c8fec5b87d8ebdb723/src/WCSimWCDigitizer.cc#L355) and [L405](https://github.com/tdealtry/WCSim/blob/0b06acf4572ade3915e697c8fec5b87d8ebdb723/src/WCSimWCDigitizer.cc#L405) of [`src/WCSimWCDigitizer.cc`](https://github.com/tdealtry/WCSim/blob/0b06acf4572ade3915e697c8fec5b87d8ebdb723/src/WCSimWCDigitizer.cc).  In both of these cases, the second parameter is hard-coded to `1.0`.  This new feature is not being used in the current code-base, but it makes sense to have consistent implementations of it.

So, on to the specific occurences of 'unused parameter' warnings from `tdealtry/main_into_hybrid:src/WCSimPMTObject.cc`:
- `HPD20inchHQE::HitTimeSmearing` on [L1702-L1717](https://github.com/tdealtry/WCSim/blob/0b06acf4572ade3915e697c8fec5b87d8ebdb723/src/WCSimPMTObject.cc#L1702-L1717).  This is interesting in that the resolution is not single-valued.  Further, random smearing factor that is the final result is biased---it is not symmetric about 0.  It is not obvious that a multiplicative fudge factor makes sense!  In this case, it is probably best to delete the name of the parameter and add a comment that the TTSFF parameter is not used in this implementation.
- `HPD12inchHQE::HitTimeSmearing` on [L1894-L1909](https://github.com/tdealtry/WCSim/blob/0b06acf4572ade3915e697c8fec5b87d8ebdb723/src/WCSimPMTObject.cc#L1894-L1909).  The situation is the same as that for the `HPD20inchHQE`---a complicated parameterization for which a multiplicative fudge-factor is not obviously correct.  Do the same thing---delete the name of the parameter and leave a comment.
- `BoxandLine20inchHQE::HitTimeSmearing` on [L2095-L2109](https://github.com/tdealtry/WCSim/blob/0b06acf4572ade3915e697c8fec5b87d8ebdb723/src/WCSimPMTObject.cc#L2095-L2109).  *ibid*.
- `BoxandLine12inchHQE::HitTimeSmearing` on [L2293-L2308](https://github.com/tdealtry/WCSim/blob/0b06acf4572ade3915e697c8fec5b87d8ebdb723/src/WCSimPMTObject.cc#L2293-L2308).  *ibid*.
- `PMT4inchR12199_02::HitTimeSmearing` on [L2725-L2739](https://github.com/tdealtry/WCSim/blob/0b06acf4572ade3915e697c8fec5b87d8ebdb723/src/WCSimPMTObject.cc#L2725-L2739).  *ibid*.
- `PMT5inchR12199_02::HitTimeSmearing` on [L2919-L2933](https://github.com/tdealtry/WCSim/blob/0b06acf4572ade3915e697c8fec5b87d8ebdb723/src/WCSimPMTObject.cc#L2919-L2933).  *ibid*.
- `PMT3inchR14374::HitTimeSmearing` on [L3116-L3125](https://github.com/tdealtry/WCSim/blob/0b06acf4572ade3915e697c8fec5b87d8ebdb723/src/WCSimPMTObject.cc#L3116-L3125).  This is an interesting case in which the second parameter `TTSFF` is used, but the *first* parameter `Q` is not.  The `timingResolution` for this implementation is not charge-dependent.  However, the same method of suppressing the warning makes sense---delete the name of the parameter and leave an explanatory comment.

**Comment by spradlin** [comment]
## Warnings from WCSimSteppingAction.cc

The warnings here are the same as those [noted in my Issue 9](https://github.com/spradlin/WCSim/issues/9#issuecomment-1002298301).  Copying the relevant discussion from there below:

### Conflicts with named units s and g

Most of the warnings are due to use of local variables named `s` and `g` that shadow the global named constants for physical units of second and gram (see https://github.com/spradlin/WCSim/issues/9#issuecomment-1001859817). In these cases, the use of the local variable has nothing to do with the physical unit. Instead, they are one-letter stand-ins for a type, like `string` or `TGraph`.

These appear to be occurring because the `G4SIunits.hh` header file is included before the header files for the ROOT components, which has the result of bringing the unit constants into the global namespace to conflict with the inlines that are defined in the Root headers.

It surprises me that the `G4SIunits.hh` constants aren't encapsulated in a namespace.

These warnings can be avoided just by rearranging the `#include` statements, so i did.

**Comment by spradlin** [comment]
## Warnings from WCSimWCAddDarkNoise.cc

There is one `uninitialized` parameter warning in the compilation of [`src/WCSimWCAddDarkNoise.cc`](https://github.com/tdealtry/WCSim/blob/0b06acf4572ade3915e697c8fec5b87d8ebdb723/src/WCSimWCAddDarkNoise.cc), specifically in the function [`WCSimWCAddDarkNoise::AddDarkNoiseBeforeDigi()`](https://github.com/tdealtry/WCSim/blob/0b06acf4572ade3915e697c8fec5b87d8ebdb723/src/WCSimWCAddDarkNoise.cc#L173-L336).  The variable `thenum_pmts` used in [L187](https://github.com/tdealtry/WCSim/blob/0b06acf4572ade3915e697c8fec5b87d8ebdb723/src/WCSimWCAddDarkNoise.cc#L187).  `thenum_pmts` is declared at [L182](https://github.com/tdealtry/WCSim/blob/0b06acf4572ade3915e697c8fec5b87d8ebdb723/src/WCSimWCAddDarkNoise.cc#L182).  It is conditionally initialized in [L183-L186](https://github.com/tdealtry/WCSim/blob/0b06acf4572ade3915e697c8fec5b87d8ebdb723/src/WCSimWCAddDarkNoise.cc#L183-L186).  The final `else` does not include an assignment for `thenum_pmts`, so if something is wrong it may be unitialized.  (If execution gets to the final `else`, something is very wrong---an throwing an exception is probably the right thing to do.)

Without more substantially rewriting `WCSimWCAddDarkNoise::AddDarkNoiseBeforeDigi()`, it is probably best and safest to initialize `thenum_pmts` to 0 at declaration.

**Comment by spradlin** [comment]
## Warnings from WCSimWCSD.cc

These are similar to [those from `WCSimSteppingAction.cc`](https://github.com/spradlin/WCSim/issues/16#issuecomment-1374069533).  They look like conflicts caused by poor encapsulation in GEANT4 and ROOT.  It looks like these conflicts can be generally avoided by ordering header `#include` statements by
```
   WCSim component headers
              v
    ROOT component headers
              v
  GEANT4 component headers
```
Rearranging the `#include`s in [`src/WCSimWCSD.cc`](https://github.com/tdealtry/WCSim/blob/0b06acf4572ade3915e697c8fec5b87d8ebdb723/src/WCSimWCSD.cc#L1-L19) to put the [WCSim headers (L17-L19)](https://github.com/tdealtry/WCSim/blob/0b06acf4572ade3915e697c8fec5b87d8ebdb723/src/WCSimWCSD.cc#L17-L19) before the [G4 headers (L2-L13)](https://github.com/tdealtry/WCSim/blob/0b06acf4572ade3915e697c8fec5b87d8ebdb723/src/WCSimWCSD.cc#L2-L13) avoids these warnings.

**Comment by spradlin** [comment]
Resolved with merge of https://github.com/tdealtry/WCSim/pull/1.
